### ISSUE TITLE
Receiving error UNEXPECTED_FRAME - expected content header for class 60, got non content header frame instead

### ISSUE THREAD

**Issue by bilmich1** [CopernicaMarketingSoftware/AMQP-CPP#394]
Relates to #92, but decided to open a new issue considering the other one is almost 5 years old at this point.

Hello everyone and thanks in advance for the help. I am currently working on implementing the RabbitMQ message broker into my application but ran into the problem that eventually, without fail, I'll get the following error (after a few hundred messages):
`UNEXPECTED_FRAME - expected content header for class 60, got non content header frame instead` 

Minimum example repo: https://github.com/bilmich1/amqpcpp-test
Tested using AMQP-CPP 4.3.11 and boost 1.75.0, both pulled via vcpkg as well as RabbitMQ 3.8.9 (Erlang 23.2.1) running in a docker container.

Why the need for Synchronous operation? you might ask. We aim to support multiple message broker technologies and the common interface requires it (not shown in the repo linked above).  

Output example (contains a successful publish followed by one with the error):
```
AmqpCppStreamer::publish begin
SynchronousChannel::publish begin
SynchronousChannel::publish locked
AsioHandler::doWrite begin
AsioHandler::doWrite async_write
SynchronousChannel::waitForOperationToFinish begin
AsioHandler::doWrite begin
AsioHandler::doWrite async_write
AsioHandler::doWrite begin
AsioHandler::doWrite async_write
AsioHandler::doRead async_receive
SynchronousChannel::publish onAck
AsioHandler::doRead begin
SynchronousChannel::waitForOperationToFinish finished
AmqpCppStreamer::publish success

AmqpCppStreamer::publish begin
SynchronousChannel::publish begin
SynchronousChannel::publish locked
AsioHandler::doWrite begin
AsioHandler::doWrite async_write
AsioHandler::doWrite begin
AsioHandler::doWrite async_write
AsioHandler::doWrite begin
SynchronousChannel::waitForOperationToFinish begin
AsioHandler::doWrite async_write
AsioHandler::doRead async_receive
AsioHandler::doWrite begin
SynchronousChannel::publish onLost
SynchronousChannel::waitForOperationToFinish finished
AsioHandler::onError: UNEXPECTED_FRAME - expected content header for class 60, got non content header frame instead
SynchronousChannel::waitForOperationToFinish error: SynchronousChannel error: Message failed to publish!
AmqpCppStreamer::runConnectionService error: AmqpCppStreamer::publish error: SynchronousChannel error: Message failed to publish!
UNEXPECTED_FRAME - expected content header for class 60, got non content header frame instead
Exception in main: SynchronousChannel error: Message failed to publish!
Done
AmqpCppStreamer::stop begin
AmqpCppStreamer::stop joined
AmqpCppStreamer::stop reset
```

**Comment by EmielBruijntjes** [comment]
The objects in AMQP-CPP are not thread-safe. If you send instructions in one thread, while you process messages in another, you will eventually end up in these sort of errors. If this is a real issue, you must send in a way smaller piece of code in which you dont start any threads.

**Comment by bilmich1** [comment]
As I was trying to come up with a comment that described my misunderstanding on how the lib is supposed to work, I worked myself an idea to fix my issue. I guess I need to thank you for acting as my debugging duck.

For anyone wondering, and I feel foolish for not thinking about it earlier, the solution was simply to queue the publish operation on the event queue (so that it's executed by the loop thread) instead of calling publish on the channel object from the second thread. I have updated the code in the linked repository above for posterity if someone five years from now finds this issue like I did the other one.
